sg_jira/handlers/entity_issue_handler.py

```diff
@@ -26,6 +38,14 @@ def __init__(self, syncer, issue_type):
         super(EntityIssueHandler, self).__init__(syncer)
         self._issue_type = issue_type
 
+        # Due to GDPR, some changes were done to JIRA Cloud which complicates
+        # matching users by email. So let's use the right resolver based
+        # on the server type.
+        if self._jira.is_jira_cloud:
```

Comment:
This is the crux of the fix for JIRA cloud. We use the right method to map users depending on which server type we're talking to.